src/grt/src/Grid.cpp

```diff
@@ -79,9 +79,14 @@ void Grid::clear()
 
 void Grid::addObstruction(int layer, const odb::Rect& obstruction)
 {
-  odb::Rect obs_inside_die = die_area_.intersect(obstruction);
-  if (!obs_inside_die.isInverted()) {
-    obstructions_[layer].push_back(obs_inside_die);
+  // compute the intersection between obstruction and the die area
+  // only when they are overlapping to avoid assert error during
+  // intersect() function
+  if (die_area_.overlaps(obstruction)) {
```

Comment:
Do you want to include the edge of the shapes or not? `overlaps()` will exclude the edge but `intersects()` will include it.

Reply:
I think I don't want to include the edge of the shape. I want to compute the intersection of the two rectangles only when this intersection results in a non-zero area rectangle.